### PR TITLE
Puppetv4

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -3,3 +3,21 @@ begin
   require 'rspec-system/rake_task'
 rescue LoadError
 end
+
+require 'puppet-lint/tasks/puppet-lint'
+PuppetLint.configuration.send('disable_80chars')
+PuppetLint.configuration.relative = true
+PuppetLint.configuration.ignore_paths = ["spec/**/*.pp", "pkg/**/*.pp"]
+
+desc 'Validate manifests, templates, and ruby files'
```
Comment (ekohl) [inline Rakefile]: There's a [validate task in puppetlabs_spec_helper](https://github.com/puppetlabs/puppetlabs_spec_helper/blob/master/lib/puppetlabs_spec_helper/rake_tasks.rb). Why not use that?

**Comment by ghoneycutt** [comment]
@ekohl I removed that bit from the Rakefile and rebased. Ready for merge?

**Comment by ekohl** [comment]
:+1: on green tests.

**Comment by ghoneycutt** [comment]
@DavidS Could you please review this PR?

File: Gemfile [hunk 2]
```diff
@@ -23,4 +23,9 @@ else
   gem 'puppet', :require => false
 end
 
+# rspec must be v2 for ruby 1.8.7
+if RUBY_VERSION >= '1.8.7' and RUBY_VERSION < '1.9'
+  gem 'rspec', '~> 2.0'
```
Comment (DavidS) [inline Gemfile]: rspec 3.1 is the last rspec that works on ruby 1.8.7:

https://github.com/puppetlabs/puppetlabs-stdlib/blob/master/Gemfile#L15

Comment (DavidS) [inline Gemfile]: I do like the basic formulation.

File: .travis.yml [hunk 3]
```diff
@@ -10,16 +10,29 @@ rvm:
...
     - PUPPET_GEM_VERSION="~> 3.1.0"
     - PUPPET_GEM_VERSION="~> 3.2.0"
+    - PUPPET_GEM_VERSION="~> 3.3.0"
     - PUPPET_GEM_VERSION="~> 3.4.0"
-    - PUPPET_GEM_VERSION="~> 3.5.1"
+    - PUPPET_GEM_VERSION="~> 3.5.0"
     - PUPPET_GEM_VERSION="~> 3.6.0"
     - PUPPET_GEM_VERSION="~> 3.7.0"
     - PUPPET_GEM_VERSION="~> 3.8.0"
+    - PUPPET_GEM_VERSION="~> 3" PARSER="future"
```
Comment (DavidS) [inline .travis.yml]: You can just set FUTURE_PARSER="yes" and puppetlabs_spec_helper will pick it up and configure it for you. Then you can drop off the change to spec_helper

**Comment by DavidS** [comment]
Please squash the two commits pertaining to the nice parameter.

File: .travis.yml (hunk 3 above)
Comment (ghoneycutt) [inline .travis.yml]: Right on. I just took this from another Puppet Labs project to get it working.

Comment (DavidS) [inline .travis.yml]: I do need to make time to go through all modules and stamp this out.

http://media.tumblr.com/tumblr_m5ozx7iP3x1qcwic6.gif

**Comment by DavidS** [comment]
Oh wow, this module does still use the serverspec things :-/

**Comment by ghoneycutt** [comment]
So changing rspec from `~> 2.0` to `~> 3.1.0` seems to be causing the build failure for ruby 1.8.7. Mind if I revert that so we can get this merged?

**Comment by ghoneycutt** [comment]
That might of worked with stdlib though that module is fundamentally different

**Comment by DavidS** [comment]
Thank you for your contribution!

**Comment by ghoneycutt** [comment]
@DavidS thanks for the review

**Pull request merged**